### ISSUE TITLE
Some comments may be missed in threaded comments

### ISSUE THREAD

**Issue by mp31415** [jarednova/timber#713]
In timber-post.php foreach loop at line 737 a comment tree structure is built:
```
foreach( $timber_comments as $key => $comment ) {
	if ( $comment->is_child() ) {
		unset($timber_comments[$comment->id]);
		if ( isset($timber_comments[$comment->comment_parent]) ) {
			$timber_comments[$comment->comment_parent]->children[] = $comment;
		}
	}
}
```
Unfortunately there is a bug in this loop. When a child is detected and removed from $timber_comments it may be a parent of another comment. When that comment is found down the road there will be no parent anymore in $timber_comments list and it will be silently ignored. The sequence of events probably depends on the comment query result, but the issue is real. I would suggest the following minor change to the loop above:
```
$child_comment_ids = [];
foreach( $timber_comments as $key => $comment ) {
	if ( $comment->is_child() ) {
		//unset($timber_comments[$comment->id]);
		$child_comment_ids[] = $comment->id;

		if ( isset($timber_comments[$comment->comment_parent]) ) {
			$timber_comments[$comment->comment_parent]->children[] = $comment;
		}
	}
}
		
foreach($child_comment_ids as $id) {
	unset($timber_comments[$id]);
}
```
The ids of children are collected but they are removed from $timber_comments only after the first loop. It fixed the problem for me.

**Comment by oxyplay** [comment]
Same problem, trying to make nested comments and receive only two-level array. 
Commented timber-post.php lines 766-773 and used suggested code below - everything works!

**Comment by connorjburton** [comment]
This seems to actually have been fixed in #780
